### PR TITLE
Issue 217: restrict self hosted to known users

### PR DESCRIPTION
**Pull request by kkaempf**
Run `main` CI task on self-hosted runners for known users

Also bump GINGKO_NODES to 6. The current tests don't make use of more than 6 nodes.

This brings the CI time down to approx. 8 minutes.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -8,7 +8,8 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: self-hosted
+    if: contains(["kkaempf", "jimmikarily", "andreas-kupries", "rohitsakala", "svollath", "thardeck"], "${{ github.actor }}")
```
Comment (jimmykarily) [inline .github/workflows/main.yml:12]:
```suggestion
    if: contains(["kkaempf", "jimmykarily", "andreas-kupries", "rohitsakala", "svollath", "thardeck"], "${{ github.actor }}")
```

**Comment by jimmykarily** [comment]
Can this be combined with the labeling method inplemented in #219 ? As in "run tests if the actor is known or someone labeled the PR".

Also, do we know who the "actor" is? Is it the committer of the last commit, the person opening the PR or someone else? Who is the actor when we simply merge to master?

File: .github/workflows/main.yml (hunk 1 above)
Comment (kkaempf) [reply .github/workflows/main.yml:12]: :facepalm: :wink:

**Comment by kkaempf** [comment]
The actor, in the github action sense, is the one triggering the action. As `workflow/main.yml` triggers on `push` and `pull_request`, we _should_ be safe.

#219 is the correct solution for _outside_ PRs, where someone with write access to the repo must explicitly add a label to trigger the action run. I haven't found a way to combine both inside `workflow/main.yml` (would need an `else` expression). So I'd guess we need another action for this. (PR follows :wink:)

**Comment by kkaempf** [comment]
#228 opened

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -8,7 +8,8 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: self-hosted
+    if: contains(["kkaempf", "jimmykarily", "andreas-kupries", "rohitsakala", "svollath", "thardeck"], "${{ github.actor }}")
```
Comment (jimmykarily) [inline .github/workflows/main.yml:12]: wouldn't this combine both #217 and #228 ? : 

```
if : contains(["kkaempf", "jimmykarily", "andreas-kupries", "rohitsakala", "svollath", "thardeck"], "${{ github.actor }}") || contains(github.event.pull_request.labels.*.name, 'safe to test')
```

https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#operators

Comment (kkaempf) [reply .github/workflows/main.yml:12]: :thinking: 

I also don't know how much importance the `types: [ labeled ]` line has.

Comment (jimmykarily) [reply .github/workflows/main.yml:12]: I'm reading [this relatively irrelevant page](https://github.com/marketplace/actions/label-actions#examples) but afaict `types: [ labeled]` refers to the PR events that will trigger the workflow. That means the workflow it only triggered when a label is added. Combined with the label check it does what we want.

Comment (kkaempf) [reply .github/workflows/main.yml:12]: Yes, `types: [labeled]` is the trigger. But this trigger would be valid for any actor. Which doesn't match my intention:
- if known actor, run unconditionally
- if unknown actor, run only when `safe to test` label is added

I fear this cannot be combined in a single action.

Comment (jimmykarily) [reply .github/workflows/main.yml:12]: Now I get it. You are right. I would still prefer to have them combined in one like this:

- Run PRs only when tagged. This allows us to skip builds when we are working on a PR, pushing commits but when we are not done yet. I don't mind manually tagging the PR again to trigger a build.
- Run main branch only when the actor is one of the team members (when I merge a PR, I'm the actor right?).

**Review by jimmykarily — APPROVED** [review]

**Pull request merged**